Source/DafnyCore/Resolver/PreType/PreTypeConstraint.cs

```diff
@@ -38,4 +38,17 @@ public abstract class PreTypeConstraint {
       this.errorFormatStringProducer = errorFormatStringProducer;
     }
   }
+
+  public abstract class OptionalErrorPreTypeConstraint : PreTypeConstraint {
```

Comment:
This class seems like premature deduplication. Please inline it.

Reply:
This common superclass lets method `ConstrainTypeArguments` in `PreTypeSubtypeConstraint.cs` handle both subtype constraints and equality constraints.

Reply:
I see. Thanks